examples/ipv6only-app/esp32/main/main.cpp

```diff
@@ -0,0 +1,650 @@
+    { 196, 4980 },
+} });
+
+constexpr size_t kUdpBufferSize = 512;
+
+constexpr uint8_t kWiFiConnectRetryMax = 5;
+
+/* The event group allows multiple bits for each event, but we only care about
+ * whether the STA initialization was successful
+ */
+constexpr uint8_t WIFI_STA_UP_BIT = BIT0;
+
+/* The event group allows multiple bits for each event, but we only care about 3 events:
+ * - we are connected to the AP with an IP6
+ * - we are connected to the AP with an IP4
+ * - we failed to connect after the maximum amount of retries */
+#define WIFI_CONNECTED_IP4_BIT BIT1
+#define WIFI_CONNECTED_IP6_BIT BIT2
+#define WIFI_FAIL_BIT BIT3
+
+SemaphoreHandle_t wifi_init_semaphore;
+SemaphoreHandle_t scanresults_semaphore;
+
+/* FreeRTOS event group to signal when we are connected*/
+EventGroupHandle_t s_wifi_event_group;
+
+int s_retry_num        = 0;
+esp_netif_t * wifi_sta = nullptr;
+
+int sockfd = 0;
+unsigned int clientlen;         // byte size of client's address
+struct sockaddr_in6 clientaddr; // client addr
+
+::pw::rpc::ServerWriter<chip_rpc_ScanResults> scanresult_writer;
+
+class RAIIGuard
+{
+    std::function<void()> release_;
+
+public:
+    RAIIGuard(std::function<void()> acquire, std::function<void()> release) : release_(release) { acquire(); }
+    virtual ~RAIIGuard() { release_(); }
+};
+
+void wifi_start_event_handler(void * arg, esp_event_base_t event_base, int32_t event_id, void * event_data)
+{
+    if (event_base == WIFI_EVENT && event_id == WIFI_EVENT_STA_START)
+    {
+        ESP_LOGI(TAG, "EVENT: WIFI_EVENT_STA_START");
+        xEventGroupSetBits(s_wifi_event_group, WIFI_STA_UP_BIT);
+    }
+}
+
+void wifi_conn_event_handler(void * arg, esp_event_base_t event_base, int32_t event_id, void * event_data)
+{
+    if (event_base == WIFI_EVENT && event_id == WIFI_EVENT_STA_DISCONNECTED)
+    {
+        ESP_LOGI(TAG, "EVENT: WIFI_EVENT_STA_DISCONNECTED, reason: %d",
+                 (static_cast<system_event_sta_disconnected_t *>(event_data))->reason);
+        if (s_retry_num < kWiFiConnectRetryMax)
+        {
+            esp_wifi_connect();
+            s_retry_num++;
+            ESP_LOGI(TAG, "retry to connect to the AP");
+        }
+        else
+        {
+            xEventGroupSetBits(s_wifi_event_group, WIFI_FAIL_BIT);
+            (static_cast<chip_rpc_ConnectionResult *>(arg))->error =
+                chip_rpc_CONNECTION_ERROR((static_cast<system_event_sta_disconnected_t *>(event_data))->reason);
+        }
+    }
+    else if (event_base == WIFI_EVENT && event_id == WIFI_EVENT_STA_CONNECTED)
+    {
+        ESP_ERROR_CHECK(esp_netif_create_ip6_linklocal(wifi_sta));
+        ESP_LOGI(TAG, "Connected, link local address created");
+    }
+    else if (event_base == IP_EVENT && event_id == IP_EVENT_STA_GOT_IP)
+    {
+        auto * event = static_cast<ip_event_got_ip_t *>(event_data);
+        ESP_LOGI(TAG, "got ip4: " IPSTR, IP2STR(&event->ip_info.ip));
```

Comment:
Do you see this log at all when testing? It shouldn't occur on ipv6-only app right?

Reply:
Correct, this print doesn't show up:
```
I (104064) ipv6only: Connected, link local address created
I (105574) ipv6only: got ip6: fe80:0000:0000:0000:266f:28ff:fe8f:f3c0
I (105574) ipv6only: UDP server (re)starting
I (105574) ipv6only: connected to ap SSID:MrWifi password:****
I (105574) ipv6only: UDP server bound to port 8765
Out[5]: (Status.OK, chip.rpc.ConnectionResult())

```

Reply:
The log was added so that if it shows up in your logs you know that something is wrong, and the device is getting an IP4 address. May be removed if we are absolutely sure that can not happen.